### ISSUE TITLE
McStas + Mantid on windows, seems to only work via mcdisplay.pl

### ISSUE THREAD

**Issue by willend** [McStasMcXtrace/McCode#459]
On Windows 10, when IDF xml file is generated via mcdisplay-mantid (py), it is not picked up 100% correctly by the subsequent mcrun --format=NeXus.

Mantid barks:
![mantidplotscreensnapz003](https://cloud.githubusercontent.com/assets/8593355/25735530/530c9656-316c-11e7-9c55-d100121e2220.png)

The XML itself loads correctly in Mantid using LoadEmptyInstrument, and is also present inside the HDF file:
![hdfviewscreensnapz001](https://cloud.githubusercontent.com/assets/8593355/25735586/c168a63a-316c-11e7-8e74-a1891d7c1064.png)

I think the issue may be the highlighted filename ^ (red circle) that probably needs to match the name of the binary, in this case ILL_H16_IN5_Mantid.exe

I think this one is not important enough that it should stop us from releasing, but it would be good if you had a look @climbcat . - Idea: Assumption that instrument name is something.out? And hence nothing filled in the template?

**Comment by willend** [comment]
@trnielsen could you try to check how well this works / does not work in its current state

**Comment by climbcat** [comment]
The current version of VMWare needs to do some updates, but the auto-update mechanism fails with this (from the log file) which prevents compilation: 

<pre>
2018-10-16T13:13:44.838+02:00| vthread-1| I125: Failed to find /lib/modules/4.15.0-34-generic/build/include/linux/version.h
</pre>

**Comment by willend** [comment]
Hi there,

You are likely missing the Linux kernel headers,  try

<pre>
sudo apt-get install linux-headers-generic
</pre>

and have another go.

Best and greetings from a plane ✈️ over the Atlantic.

Peter
________________________________
Fra: Jakob Garde <notifications@github.com>
Sendt: tirsdag, oktober 16, 2018 7:19 AM
Til: McStasMcXtrace/McCode
Cc: Peter Kjær Willendrup; Assign
Emne: Re: [McStasMcXtrace/McCode] McStas + Mantid on windows, seems to only work via mcdisplay.pl (#459)


The current version of VMWare needs to do some updates, but the auto-update mechanism fails with this (from the log file) which prevents compilation:

2018-10-16T13:13:44.838+02:00| vthread-1| I125: Failed to find /lib/modules/4.15.0-34-generic/build/include/linux/version.h

**Comment by climbcat** [comment]
Have a nice journey!

**Comment by willend** [comment]
@trnielsen while we wait for @ebknudsen and I to do something about #689, could you possibly proceed by evaluating if (most of) the geometry information is created correctly using the mcdisplay-mantid Python script? 

(My expectation is that it close to works, but we should ensure that the @ebknudsen patches done in 
https://github.com/McStasMcXtrace/McCode/commit/b2c99a8bf9c4e39381f800379fad37f900d6e941#diff-274030ef2e9c5e657607e41533668fa4 and
https://github.com/McStasMcXtrace/McCode/commit/b0dd14f7cc0de76bc89827eb86414ff4ad260727#diff-274030ef2e9c5e657607e41533668fa4
are ported to the Python version also)

**Comment by willend** [comment]
Have attempted to port the changes from 
https://github.com/McStasMcXtrace/McCode/commit/b0dd14f7cc0de76bc89827eb86414ff4ad260727#diff-274030ef2e9c5e657607e41533668fa4 over to Python by https://github.com/McStasMcXtrace/McCode/commit/ab0c247a0e2ca13885a7c2123ae3db92516f277e.

Currently my Mantid crashes on visualising the instrument geometry outputting a resulting dataset. We should consider if this is worth pursuing this side of release!

**Comment by ebknudsen** [comment]
A thought: possibly this relates to the 1-off error fixed by b2c99a8. I can't say for sure though. IIRC I fixed the issues in the opposite order for mcdisplay.pl.

**Comment by willend** [comment]
Right, could easily be. Let's have a look next week during release camp? (And decide if we do anything about it there - there is a solution if the Perl junk is used)

**Comment by willend** [comment]
A new set of binaries are available for test at http://tmp.mccode.org/Windows/

**Comment by willend** [comment]
Conclusion?

**Comment by willend** [comment]
@trnielsen if you would eventually have a look here it would be great :-)
